### PR TITLE
Destination postgres (dv2): add indexes to raw table

### PR DESCRIPTION
**Pull request by edgao**
copied from https://github.com/airbytehq/airbyte/pull/34225, but without screwy merge. Still targeting the https://github.com/airbytehq/airbyte/pull/34177 branch. 

There was still one open discussion:

@evantahler 
> ... should JdbcDestinationHandler detect index changes? We got that working in snowflake. It seems like a feature we'll want in the future maybe?

@edgao 

> I think eventually yes, but not literally today. If any beta users want us to change things indexes, I feel ok asking them to manually drop all their tables? Having autodetections feels like a "after we release to GA and we want to change our indexes" feature
> it's probably not a huge lift (something like fetch jdbc metadata -> dump into the TableDefinition struct -> diff appropriately) and is a pure bolt-on addition in the future, so I feel ok deferring it for now?

### CODE REVIEW

**Review by evantahler — APPROVED** [review]

File: airbyte-integrations/connectors/destination-postgres-strict-encrypt/metadata.yaml [hunk 1]
```diff
@@ -19,8 +13,14 @@ data:
     normalizationIntegrationType: postgres
     normalizationRepository: airbyte/normalization
     normalizationTag: 0.4.1
+  registries:
+    cloud:
+      dockerImageTag: 0.4.0
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-postgres-strict-encrypt/metadata.yaml:18]: I think we should remove this, even if it's ignored

File: airbyte-integrations/connectors/destination-redshift/build.gradle [hunk 2]
```diff
@@ -4,7 +4,7 @@ plugins {
 }
 
 airbyteJavaConnector {
-    cdkVersionRequired = '0.12.1'
+    cdkVersionRequired = '0.13.0'
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-redshift/build.gradle:7]: might as well bump to 0.13.1?

**Comment by walker-philips** [comment]
@edgao What was the naming convention for indexes being created in the airbyte_internal tables? Was it create index on  <tbl name>_airbyte_raw_id or  <tbl name>_pkey by chance?

I am seeing a massive amount of essentially duplicated indexes and both are being actively created on full refresh overwrite syncs which indicates its not legacy indexes from early Airbyte versions

**Comment by edgao** [comment]
the airbyte_internal tables create three indexes https://github.com/airbytehq/airbyte/pull/34236/files#diff-b3234783bac9e700f1c1ae9155feafb81c0574ebb015ee5db297b3dc76effcb4R37-R39

which should be suffixed with `_raw_id`, `_extracted_at`, and `_loaded_at`. But also, they're `create if not exists`, so it's surprising that they're being duplicated...

**Comment by walker-philips** [comment]
Sorry, I mean massive in the sense it applies to almost every single Postgres Airbyte internal table. Our monitoring tool is reporting 700+ duplicate indexes

the two indexes that are duplicating:
<stream name>_pkey: uses a unique btree index on _airbyte_raw_id
<stream name>_raw_id: regular btree index on _airbyte_raw_id

I would argue there is no purpose for the raw_id index if the pkey index is created. Additionally a unique index on a full refresh overwrite table may have limited benefits, but maybe a simple check on the raw_id index to see if a "pkey" index exists then do not create it would suffice? I suppose that relies on _when_ the indexes are being created though. 

I am pretty sure I found the code that generates the three indexes you mentioned, but I couldn't find the "pkey" index creation.

I created [this issue](https://github.com/airbytehq/airbyte/issues/61349) with some additional details. We are on Airbyte 1.5.1 and Postgres 2.4.4

**Pull request merged**